src/open-uri.c

```diff
@@ -713,21 +713,26 @@ handle_open_in_thread_func (GTask *task,
           /* Launch the app directly */
           g_autoptr(GError) error = NULL;
 
-          g_debug ("Skipping app chooser");
-
           gboolean result = launch_application_with_uri (app, uri, parent_window, writable, &error);
-          if (request->exported)
+          if (result)
```

Comment:
This means that you also won't skip the app chooser if we fail to export files. Which is probably not the intended behaviour.

How about:
```C
gboolean
app_exists (const char *app_id)
{
[...]
}
```
And around `if (app)`:
```C
gboolean app_exists;

app_exists = app_exists (app);
if (app && app_exists) {
[...]
} else if (!app_exists) {
  g_debug ("Would have used '%s' to skip app chooser, but does not exist", app);
}
```

Add add a g_warning and a return in `launch_application_with_uri()`.

Reply:
> This means that you also won't skip the app chooser if we fail to export files. Which is probably not the intended behaviour.

No, we really do skip the app chooser when not exported. It's the next condition down. OK?

I added a g_debug() to warn when the application doesn't exist. I don't think it's worth g_warning() because it's expected that this will happen all the time.

Reply:
I've reworked this, because I didn't like that we were waiting until the last minute to make those checks. I've kept the app info creation check, and I've added those changes to:
https://github.com/flatpak/xdg-desktop-portal/pull/473